src/lib/components/chat/Messages.svelte

```diff
@@ -223,32 +223,21 @@
 		}, 100);
 	};
 
-	// TODO: change delete behaviour
-	// const deleteMessageAndDescendants = async (messageId: string) => {
-	// 	if (history.messages[messageId]) {
-	// 		history.messages[messageId].deleted = true;
-
-	// 		for (const childId of history.messages[messageId].childrenIds) {
-	// 			await deleteMessageAndDescendants(childId);
-	// 		}
-	// 	}
-	// };
-
-	// const triggerDeleteMessageRecursive = async (messageId: string) => {
-	// 	await deleteMessageAndDescendants(messageId);
-	// 	await updateChatById(localStorage.token, chatId, { history });
-	// 	await chats.set(await getChatList(localStorage.token));
-	// };
-
 	const messageDeleteHandler = async (messageId) => {
-		if (history.messages[messageId]) {
-			history.messages[messageId].deleted = true;
+		const messageParentId = history.messages[messageId]?.parentId;
 
-			for (const childId of history.messages[messageId].childrenIds) {
-				history.messages[childId].deleted = true;
-			}
+		if (messageParentId !== null) {
```

Comment:
Thanks for the PR, correct me if I'm wrong but shouldn't this part only delete the provided messageId from the list? (e.g. `history.messages[messageParentId].childrenIds.filter((id) => id !== messageId)`)

Reply:
`history.messages[messageParentId].childrenIds = []` 
This will delete all subsequent messages in addition to the deleted message itself which is what we want right?